lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -161,6 +169,12 @@ internal sealed class IMRUDriver<TMapInput, TMapOutput, TResult, TPartitionType>
             _serviceAndContextConfigurationProvider =
                 new ServiceAndContextConfigurationProvider<TMapInput, TMapOutput, TPartitionType>(dataSet, configurationManager);
 
+            if (lifecycleManager != null)
+            {
+                var handle = lifecycleManager.Subscribe(this as IObserver<IJobCancelled>);
+                _disposableResources.Add(handle);
```

Comment:
Actually does it really matter to remember the handles and call their Dispose() in the end? Because the Dispose() in this case just do unsubscribe. I think the important thing is to call Dispose() of JobLifeCycleManager to make sure Timmer is killed.